### PR TITLE
Matlab group support training

### PR DESCRIPTION
**Pull request by sbesson**
This PR rebases the additional commits included in in https://github.com/openmicroscopy/openmicroscopy/pull/4111 to test the group support via the training examples.

To test this PR, check https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-training/ is green

--rebased-from https://github.com/openmicroscopy/openmicroscopy/pull/4111

### CODE REVIEW

**Comment by sbesson** [comment]
See https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-training/83/consoleFull

File: examples/Training/matlab/ConnectToOMERO.m [hunk 1]
```diff
@@ -53,19 +53,27 @@
     delete(t);
     
     %% Admin service
-    % List groups the user is member of
+    % Retrieve  the identifier of the groups the user is member/owner of
```
Comment (mtbc) [inline examples/Training/matlab/ConnectToOMERO.m]: double space

File: examples/Training/matlab/ReadData.m [hunk 2]
```diff
@@ -24,153 +24,163 @@
...
+    % groups
+    disp('Retrieving projects owned by the session user across groups')
+    projects = getProjects(session, 'group', -1);
+    fprintf(1, '  Found %g projects\n', numel(projects));
+    for i = 1 : numel(projects),
+        print_object(projects(i));
+    end
+    fprintf(1, '\n');
+
+    % Retrieve all the unloaded projects owned by the any user in the
```
Comment (mtbc) [inline examples/Training/matlab/ReadData.m]: "the any" (also below)

File: examples/Training/matlab/ReadData.m [hunk 3]
```diff
@@ -24,153 +24,163 @@
...
+    % Retrieve all the images contained in a given dataset.
+    fprintf(1, 'Retrieving images contained in dataset %g\n', datasetId);
+    images2 = getImages(session, 'dataset', datasetId);
+    fprintf(1, '  Found %g images\n', numel(images2));
+    for i = 1 : numel(images2),
+        print_object(images2(i));
+    end
+    fprintf(1, '\n');
+
+    % Retrieve all the images contained in a given dataset.
```
Comment (mtbc) [inline examples/Training/matlab/ReadData.m]: project, not dataset?

File: examples/Training/matlab/ReadData.m [hunk 4]
```diff
@@ -188,87 +198,87 @@
...
+    % Retrieve all the plates owned by the session user.
+    disp('Listing plates owned by the session user');
+    allPlates = getPlates(session);
+    fprintf(1, '  Found %g plates\n', numel(allPlates));
+    for i = 1 : numel(allPlates),
+        print_object(allPlates(i));
+    end
+    fprintf(1, '\n');
+
+    % Retrieve all the pldates owned by the session user across all groups
```
Comment (mtbc) [inline examples/Training/matlab/ReadData.m]: plates

**Comment by sbesson** [comment]
Thanks. Since this is a rebase of rebases and other work is planned for the OMERO.matlab group support, I'll handle the comments in a follow-up PR

**Comment by mtbc** [comment]
Fine with me, thank you.

**Comment by jburel** [comment]
It will be also good to have example with group usage on http://www.openmicroscopy.org/site/support/omero5.2-staging/developers/Matlab.html

File: examples/Training/matlab/ReadData.m (hunk 3 above)
Comment (jburel) [inline examples/Training/matlab/ReadData.m]: ```disp('Retrieving projects across by any user in the current group')```
Not clear

**Comment by sbesson** [comment]
@jburel: agreed and already captured as part of https://trello.com/c/gSKqCAe2/49-omero-matlab-group-support.

**Comment by jburel** [comment]
Job is green
Comments fixes will be updated in a follow-up PR.

Merging

**Comment by sbesson** [comment]
--rebased-from #4111

**Pull request merged**